src/java/org/apache/cassandra/service/accord/AccordService.java

```diff
@@ -399,9 +399,10 @@ else if (images.isEmpty()) // First boot, single-node cluster
             int waitSeconds = 5;
             while (true)
             {
+                Epoch await = Epoch.create(configService.currentEpoch());
```

Comment:
should we use `max(metadata.epoch(), configService.currentEpoch())`?

Reply:
any idea why this was done? we have this block above

```
                metadata = ClusterMetadata.current();
                highestKnown = configService.currentEpoch();
                if (metadata.epoch.getEpoch() > highestKnown)
                {
                    remote = fetchTopologies(highestKnown + 1);
                    if (remote != null)
                        remote.forEach(configService::reportTopology, highestKnown + 1, Integer.MAX_VALUE);
                }
```

so, `if (metadata.epoch.getEpoch() > highestKnown)` "should" try to fetch topologies and load them, so the only time i see  `metadata.epoch` not being fine is if that loaded nothing? which could imply we spoke to the "wrong" nodes about this?

Reply:
I am +1 to `max(metadata.epoch(), configService.currentEpoch())` but worried about only using `configService.currentEpoch()`.

The code above should give a property that `configService.currentEpoch() >= metadata.epoch()`, and this propery should only break under 2 conditions

1) TCM knows 0 peers... in which case we are a single node cluster, so shouldn't TCM be notifying us of this epoch?
2) every single `peer` in the cluster sent a failure response back (or after several retries we gave up)... this seems problematic?